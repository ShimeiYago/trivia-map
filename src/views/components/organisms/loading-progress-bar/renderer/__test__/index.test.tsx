import { shallow, ShallowWrapper } from 'enzyme';
import { Renderer, State, Props } from '..';

let shallowWrapper: ShallowWrapper<Props, State, Renderer>;

const props: Props = {
  markersCurrentPageToLoad: 1,
  markersTotalPages: 5,
  markersFetchingState: 'waiting',
  isMobile: false,
};

describe('Shallow Snapshot Tests', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<Renderer {...props} />);
  });

  it('basic', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('on loading', () => {
    shallowWrapper.setProps({ markersFetchingState: 'loading' });
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('init of loading', () => {
    shallowWrapper.setProps({
      markersTotalPages: undefined,
      markersFetchingState: 'loading',
    });
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('with error', () => {
    shallowWrapper.setProps({
      markersFetchingState: 'error',
      markersErrorMsg: 'loading error',
    });
    shallowWrapper.setState({
      showLoadingProgressBar: true,
    });
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('mobile view', () => {
    shallowWrapper.setProps({
      isMobile: true,
    });
    shallowWrapper.setState({
      showLoadingProgressBar: true,
    });
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('componentDidUpdate', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<Renderer {...props} />);
  });

  it('should show loading progress bar if markersLoading changes to loading', () => {
    const prevProps: Props = props;
    shallowWrapper.setProps({ markersFetchingState: 'loading' });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate(prevProps);
    expect(instance.state.showLoadingProgressBar).toBeTruthy();
  });

  it('do not close loading progress bar soon even if markersLoading changes from loading', () => {
    const prevProps: Props = { ...props, markersFetchingState: 'loading' };
    shallowWrapper.setProps({ markersFetchingState: 'success' });
    shallowWrapper.setState({ showLoadingProgressBar: true });
    const instance = shallowWrapper.instance();

    instance.componentDidUpdate(prevProps);
    expect(instance.state.showLoadingProgressBar).toBeTruthy();
  });
});

describe('closeLoadingProgressBar', () => {
  beforeEach(() => {
    shallowWrapper = shallow(<Renderer {...props} />);
  });

  it('should close loading progress bar', async () => {
    shallowWrapper.setState({ showLoadingProgressBar: true });
    const instance = shallowWrapper.instance();
    instance.closeTimeMs = 0;

    await instance['closeLoadingProgressBar']();
    expect(instance.state.showLoadingProgressBar).toBeFalsy();
  });
});
