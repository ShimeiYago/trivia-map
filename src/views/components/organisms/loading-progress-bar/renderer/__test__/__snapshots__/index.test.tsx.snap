// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Snapshot Tests basic 1`] = `""`;

exports[`Shallow Snapshot Tests init of loading 1`] = `
<DialogScreen
  position="bottom"
  theme="white"
>
  <ForwardRef(LinearProgress)
    value={0}
    variant="buffer"
  />
</DialogScreen>
`;

exports[`Shallow Snapshot Tests on loading 1`] = `
<DialogScreen
  position="bottom"
  theme="white"
>
  <ForwardRef(LinearProgress)
    value={20}
    variant="buffer"
  />
</DialogScreen>
`;
